code/object/collideshipweapon.cpp

```diff
@@ -104,6 +104,17 @@ static void ship_weapon_do_hit_stuff(object *pship_obj, object *weapon_obj, vec3
 	if (wip->damage_curve_idx >= 0)
 		damage *= Curves[wip->damage_curve_idx].GetValue(f2fl(Missiontime - wp->creation_time) / wip->lifetime);
 
+	// if this is friendly fire, we check for the friendly fire cap values
+	if (wp->team == shipp->team) {
+		if (&Objects[weapon_obj->parent] == pship_obj && The_mission.ai_profile->weapon_self_damage_cap[Game_skill_level] >= 0.f) {
```

Comment:
Does this correctly handle multiple-level spawns?  E.g. if a weapon spawns a weapon that spawns another weapon?  The immediate parent would be the parent weapon, not the parent ship.

Reply:
A submunition's parent is the grandparent ship, not the spawner weapon. (When submunitions are spawned, the submunition's parent gets set to the spawner's parent, so it's consistent through any number of generations.) I think there's some specific behavior that depends on this, though I don't remember exactly what it was.